packages/payment-processor/test/payment/encoder.test.ts

```diff
@@ -224,99 +224,143 @@ beforeAll(async () => {
-    let confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-
-    tx = await wallet.sendTransaction(encodedTransactions[1]);
-    confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-  });
-
-  it('Should handle ERC20 Fee Proxy request', async () => {
-    const encodedTransactions = await encodeRequestApprovalAndPayment(
-      validRequestERC20FeeProxy,
-      provider,
-    );
-
-    let tx = await wallet.sendTransaction(encodedTransactions[0]);
-    let confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-
-    tx = await wallet.sendTransaction(encodedTransactions[1]);
-    confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-  });
-
-  it('Should handle ERC20 Conversion Proxy request', async () => {
-    let encodedTransactions = await encodeRequestApprovalAndPayment(
-      validRequestERC20ConversionProxy,
-      provider,
-      {
-        conversion: alphaConversionSettings,
-      },
-    );
-
-    let tx = await wallet.sendTransaction(encodedTransactions[0]);
-    let confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-
-    tx = await wallet.sendTransaction(encodedTransactions[1]);
-    confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-  });
-
-  it('Should handle ERC20 Swap Proxy request', async () => {
-    let encodedTransactions = await encodeRequestApprovalAndPayment(
-      validRequestERC20FeeProxy,
-      provider,
-      {
-        swap: alphaSwapSettings,
-      },
-    );
-
-    let tx = await wallet.sendTransaction(encodedTransactions[0]);
-    let confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-
-    tx = await wallet.sendTransaction(encodedTransactions[1]);
-    confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-  });
-
-  it('Should handle ERC20 Swap and Conversion Proxy request', async () => {
-    let encodedTransactions = await encodeRequestApprovalAndPayment(
-      validRequestERC20ConversionProxy,
-      provider,
-      {
-        swap: alphaSwapConversionSettings,
-        conversion: alphaConversionSettings,
-      },
-    );
-
-    let tx = await wallet.sendTransaction(encodedTransactions[0]);
-    let confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
-
-    tx = await wallet.sendTransaction(encodedTransactions[1]);
-    confirmedTx = await tx.wait(1);
-    expect(confirmedTx.status).toBe(1);
-    expect(tx.hash).not.toBeUndefined();
+const cases = ['without specific approval', 'with specific approval'];
+cases.forEach((value, index) => {
```

Comment:
that makes tests quite hard to read. 

What you should do is use [describe.each](https://jestjs.io/docs/api#describeeachtablename-fn-timeout) and have all the variables in the `table` 

something like this: 
`describe.each(["without specific approval", {approval: undefined }],["...",{approval: { ... }}])("...", ()=> { } )`

Reply:
I refactored the whole thing since there was a lot of duplicated code